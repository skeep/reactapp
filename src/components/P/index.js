// @flow

import React from 'react';

type Props = {
  content: string
};

const P = (props: Props): any => (<p>{props.content}</p>);

P.defaultProps = {
  content: "This is default content of a paragraph tag"
};

export default P;
