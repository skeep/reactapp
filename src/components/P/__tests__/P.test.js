import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import P from '../index.js';

// Snapshot test
it('Snapshot test with content', () => {
  const tree = shallow(
    <P content='content 1' />
  );
  expect(toJson(tree)).toMatchSnapshot();
});

it('Snapshot test without content prop', () => {
  const tree = shallow(
    <P />
  );
  expect(toJson(tree)).toMatchSnapshot();
});

// Rendering test
it('Rendering test with content', ()=>{
  const tree = shallow(
    <P content='content 2' />
  );
  expect(tree.find('p').text()).toEqual('content 2');
});

it('Rendering test without content prop', ()=>{
  const tree = shallow(
    <P />
  );
  expect(tree.find('p').text()).toEqual('This is default content of a paragraph tag');
});

// Behaviour tests

// Integration tests

// System tests
