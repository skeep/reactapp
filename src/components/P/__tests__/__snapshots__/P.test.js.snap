// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test with content 1`] = `
<p>
  content 1
</p>
`;

exports[`Snapshot test without content prop 1`] = `
<p>
  This is default content of a paragraph tag
</p>
`;
